wom/src/main/scala/wdl4s/wdl/WdlCall.scala

```diff
@@ -56,28 +55,37 @@ object WdlCall {
 
   private def buildWomNodeAndInputs(wdlCall: WdlCall): ErrorOr[CallWithInputs] = {
 
-    val inputToOutputPortValidations: Iterable[ErrorOr[(String, GraphNodePort.OutputPort)]] = for {
-      (inputName, expr) <- wdlCall.inputMappings
-      variable <- expr.variableReferences
-      parent <- wdlCall.parent
-      node <- parent.resolveVariable(variable.terminal.sourceString)
-      outputPort = outputPortFromNode(node, variable.terminalSubIdentifier)
-      inputNameToOutputPortMapping = outputPort map { op => inputName -> op }
-    } yield inputNameToOutputPortMapping
+    def resolveVariable(v: AstTools.VariableReference): ErrorOr[(String, GraphNodePort.OutputPort)] = {
+      val name =  v.fullVariableReferenceString
+      for {
+        parent <- wdlCall.parent.toValidNel("Call has no parent?")
+        node <- parent.resolveVariable(v.terminal.sourceString).toValidNel(s"Failed to resolve variable $name")
+        outputPort <- outputPortFromNode(node, v.terminalSubIdentifier)
+      } yield name -> outputPort
+    }
 
-    val inputToOutputPortValidation: ErrorOr[List[(String, GraphNodePort.OutputPort)]] = inputToOutputPortValidations.toList.sequence
 
-    // TODO: When WomExpressions wrap WdlExpressions, we can switch this to use expressionBasedInputs instead of portBasedInputs:
-    val combinedValidation = (inputToOutputPortValidation |@| wdlCall.callable.womDefinition) map { (_, _) }
+    val graphNodeInputExpressionValidation: ErrorOr[List[GraphNodeInputExpression]] = {
+      def graphNodeInputExpression(tuple: (String, WdlExpression)): ErrorOr[GraphNodeInputExpression] = {
```

Comment:
ToL I know you aren't a fan of returning 1-arg functions but here it could simplify tuple assignment.

```
... = {
case (name, wdlExpression) => ...
}
```